rules/prefer-flat-map.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const MESSAGE_ID = 'preferFlatMap';
+
+const isFlat = node => {
+	return (
+		node.type === 'CallExpression' &&
+		node.callee.type === 'MemberExpression' &&
+		node.callee.property.type === 'Identifier' &&
+		node.callee.property.name === 'flat'
+	);
+};
+
+const isMap = node => {
+	return (
+		node.type === 'CallExpression' &&
+		node.callee.type === 'MemberExpression' &&
+		node.callee.property.type === 'Identifier' &&
+		node.callee.property.name === 'map'
+	);
+};
+
+const report = (context, nodeFlat, nodeMap) => {
+	// Location will be:
+	//   map(...).flat();
+	//                 ^
+	const flatEnd = nodeFlat && nodeFlat.end;
+
+	// Location will be:
+	//   map(...).flat();
+	//          ^
+	const mapEnd = nodeMap && nodeMap.end;
+
+	const mapProperty = nodeMap && nodeMap.callee && nodeMap.callee.property;
+
+	context.report({
+		node: nodeFlat,
+		messageId: MESSAGE_ID,
+		fix: fixer => {
+			return [
+				fixer.removeRange([mapEnd, flatEnd]),
+				fixer.replaceText(mapProperty, 'flatMap')
+			];
+		}
+	});
+};
+
+const create = context => ({
+	CallExpression: node => {
+		if (!isFlat(node)) {
+			return;
+		}
+
+		const parent = node && node.callee && node.callee.object;
```

Comment:
Please check explicitly if `node.callee` is `MemberExpression`

Reply:
That check happens in a helper: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/284/files#diff-5b3bb9237e1bb62a17dd58af5ed9c0afR18

Reply:
But `node.callee` is not passed to `isMap`, right?

Reply:
Oh, linked the wrong helper function. https://github.com/sindresorhus/eslint-plugin-unicorn/pull/284/files#diff-5b3bb9237e1bb62a17dd58af5ed9c0afR9

The entire rule returns early if it isn't working on `.flat()` -- it can't get to the `isMap` check unless `node` is a `MemberExpression`.

Reply:
Oh, right, I see. But this line could be simplified then:
```js
const parent = node.callee.object;
```